### ISSUE TITLE
Revise Licensing Policy

### ISSUE THREAD

**Issue by odysseywestra** [mypaint/mypaint#528]
Currently our [Licensing Policy](https://github.com/mypaint/mypaint/wiki/Licensing-policy) in our wiki currently uses version 3.0 of the Creative Commons License. We should update it for version 4.0 of Creative Commons. 

[What new in Creative Commons 4.0](http://creativecommons.org/version4)

**Comment by odysseywestra** [comment]
Here's my proposed revision:


**Q:** How do I license my artwork or code for inclusion in MyPaint?

**A:** If you want your contribution to go into the official MyPaint distribution, you need to apply a license to it before we can distribute it. We use a mix of the public domain, Creative Commons licenses, and Free/Open Source Software licenses:

* **Background textures and palette files**: these need to be **public domain**
* **Brush packs**: **public domain**.
  - Alternatively, use **CC-BY (or CC-BY-SA) 4.0** for the visible parts, *but keep the raw brush settings **public domain***.
* **Supplemental artwork or promotional material**: **CC-BY (or CC-BY-SA) 4.0**
* **Program code**, icons, and artwork for display within the program: use **GNU GPLv2+** for new code; retain existing licenses when modifying stuff

The reason for this is that we try to ensure that elements which are highly likely to be reused by artists in the creation of new works, for example background texture images or raw brush settings, are made as free as possible so that artists who use them don't have to worry about copyright infringement.

### Background textures and palette files

Background texture images and palette files for inclusion in the main MyPaint distribution should be Public Domain so that artists can reuse them as freely as possible. The Creative Commons "CC0" dedication is a well-known way of advertising that. Here's what to write in your README:

> To the extent possible under law, **\<\<YOUR NAME\>\>** has waived all copyright, and related or neighboring rights to **\<<NAME(S) OF IMAGE(S)>\>**. See <http://creativecommons.org/publicdomain/zero/1.0/> for details.

Though words to that effect can be embedded in the images' metadata or comments fields instead. GIMP palette files have a description field which can be used for the same purpose. Background images and palette files you contribute may end up embedded in works that other artists wish to publish under other licenses, so this waiver is needed.

### Brush packs

Ideally, use Public Domain. Alternatively, use Creative Commons "CC-BY" or "CC-BY-SA" version 4.0, but release the raw brush settings into the Public Domain.

Long form: brush packs for inclusion into the main MyPaint distribution should be placed into the public domain just like background images, but they do not have to be. Instead you can use the Creative Commons "Attribution" or "Attribution-ShareAlike" licenses, both of which require redistributors to give you credit. Note that if you choose one of these fancier licenses, the pack's internal brush settings must still be Public Domain. Here's an example of how to say that in your brush pack's README: 

> Copyright **\<\<YEAR\>\>** **\<\<YOUR NAME\>\>**
>
> Except where otherwise noted, this work is licensed under a Creative Commons Attribution 4.0 International License. See <http://creativecommons.org/licenses/by/4.0/> for details.
>
> As an exception to this and to the extent possible under law, \<\<YOUR NAME\>\> has waived all copyright and related or neighboring rights to the raw brush settings (those files ending with ".myb"). See <http://creativecommons.org/publicdomain/zero/1.0/> for details.

We need the exception for the raw brush settings because those bits are copied around very enthusiastically by the MyPaint program — including into files others might want to distribute with their own license. Making the settings Public Domain allows them to do that. Of course, you can still claim copyright on the pack as a work as a whole, including all the preview packages and supplemental cover-type artwork in it, but excluding the raw settings.

If you want to use a Creative Commons license, it must be version 4.0, and it cannot be a NonCommercial or a NoDerivatives one.

If you choose Public Domain for the entire pack, you'll still be credited by us in the program About box unless you ask not to be named. ;) 

### Supplemental artwork or promotional material

If you want your work to be included in the main MyPaint distribution, please use one of the Creative Commons "CC-BY v4.0" or the "CC-BY-SA v4.0" licenses. For example:

> Copyright **\<\<YEAR\>\>** **\<\<YOUR NAME\>\>**
>
> This work is licenced under the Creative Commons Attribution 4.0 International License. To view a copy of this licence, visit <http://creativecommons.org/licenses/by/4.0/> or send a letter to Creative Commons, 171 Second Street, Suite 300, San Francisco, California 94105, USA.

You can express this in any form suggested by the Creative Commons guys' license chooser: <http://creativecommons.org/choose/>. Embedded metadata would be ideal. Creative Commons licenses on components of the MyPaint distribution must be version 4.0, and they cannot be a NonCommercial or a NoDerivatives license.

### Program code

All new program code and supplemental data files should be licensed under the GNU GPLv2.0, with the "or (at your option) any later version" clause. They \*must\* be licensed in a way that is compatible with this. This includes program icons and artwork for display within the program. Retain the existing license boilerplates when working on existing code.

Please put a boilerplate header on each new program file in the format recommended by the license itself. It's fine for individual programmers to claim copyright over new files which originate with them.



Beyond this issue this will also require changing the copyright files in the source code as well?

**Comment by achadwick** [comment]
Looks good to me. The proposed changes don't change anything other than the CC version number and getting rid of "unported" language, correct?

**Comment by odysseywestra** [comment]
Yup that is correct. If there isn't any more objections, I'll update the page.

**Comment by odysseywestra** [comment]
Actually before I commit this change, there one thing I need to ask. Creative Commons dose allow the ability to wavier some rights to a particular entity of their choice. It's called Creative Commons Plus.

Could we word it in a way that allows that allows the Artist to be able to license with NonCommercial and/or NonDerivative Use, but authorize MyPaint Commercial and Derivative Use.

Here's an example excerpt from CC+ WIki Page:

```html
    <span xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/">
     
    <span rel="dc:type" href="http://purl.org/dc/dcmitype/Text" property="dc:title">My Book</span> by 
    <a rel="cc:attributionURL" property="cc:attributionName" href="http://rejon.org/my_book">Jon Phillips</a> 
     
    is licensed under a 
     
    <a rel="license" href="http://creativecommons.org/licenses/by-nc/3.0/">Creative Commons 
    Attribution Non-Commercial 3.0 License</a>. 
     
    <span rel="dc:source" href="http://deerfang.org/her_book"/>
    Permissions beyond the scope of this license may be available at 
    <a rel="cc:morePermissions" 
    href="http://somecompany.com/revenue_sharing_agreement">somecompany.com</a>.
     
    </span>
```

I'll submit another revision explaining that later, and post the one we have now.

[Creative Commons Plus](https://wiki.creativecommons.org/wiki/CCPlus)

**Comment by achadwick** [comment]
@odysseywestra I always turn to the [DFSG](https://www.debian.org/social_contract#guidelines) on these matters, as a benchmark of the sort of thing that Debian and derivatives allow into their "main" repo - the freest one. It's a very real concern, because that's exactly where I want to put MyPaint! Screenshots and other promo material might be candidates for inclusion in Debian because of docs. Program artwork always would because it it's part of the app.

I think {CC-NC + not-NC-for-MyPaint} would fall foul of the discrimination clause (groups of persons) by giving *everyone other than MyPaint* a more restrictive license than members of the MyPaint team. Such a license would also fall foul of the free redistribution clause because it would prevent people downstream from selling MyPaint or collections of software which include it.

So... no on CC+, please. Special-case licenses are a huge mess that will cause problems downstream. It's clearer and simpler to state that the license must be free for everyone if work is to be included.

Our stringent requirements of free reuse™ don't affect:

* Forum postings (not even sure what, but we must gain some implicit permissions by the act of posting - it's ok for those to be special cases, for the purposes of the forum, naturally.)
* External brush packs (all we hold are the links, with CC0 or PD thumbnails on our wiki.)
* Articles about MyPaint with original screenshots (but feel free to use those our artists have released too - that's what they're for!)
* Any work made with MyPaint (the standard brush settings do not contaminate .ORA files they are included in)

**Comment by achadwick** [comment]
Updating the licensing policy raises one other question: what do we do about existing works licensed to us for inclusion that use CC licenses with version 3.0?

I think it's fine to grandfather these. After all, we've already accepted the works into MyPaint, so we've already applied a freedom test and decided they won't hurt anyone. Do we need an actual grandfather clause to say that?

**Comment by odysseywestra** [comment]
I understand. I didn't know about the DFSG so thanks for highlighting that. Probably we want to mention the [DFSG](https://www.debian.org/social_contract#guidelines) as well in case anybody asks on the forums about why we don't allow CC+ material.

I believe v3.0 is compatible to v4.0, plus each are compatible to the GPLv2 for CC-BY-SA, and CC-BY. So we can grandfather them without problem.

**Comment by odysseywestra** [comment]
From mypaint/mypaint.github.io#8
>@odysseywestra Do we always need to host the image here ourselves? If not, maybe we could suggest relaxing the license constraint to "you only have to allow us to distribute a large, good-looking thumbnail under (preferred license here)", and just linking to the artist's own gallery page for the work. Let me know if you think that's unworkable.
 ...
>Artists may only want us to use a thumbnail on our website, so they can distribute the real deal under their own license. And that's probably the best compromise position if there are any issues.

Forgot we need to write that exception in so we can accept artist's work in our Gallery Page, and the website in general on MyPaint.Org.

**Comment by odysseywestra** [comment]
Okay I've revised the draft above include content on our community site. Let me know your thoughts, and whether I can use this draft to update our current license policy.

**Comment by achadwick** [comment]
> # Content on our Community Forums

> Please understand that all content on the Community Forums are license under the CC-BY-SA v4.0. 

Suggest saying they're "automatically licensed".

I can see the lack of a NonCommercial module being unpopular. [Control over your own postings](https://en.wikipedia.org/wiki/YOYOW) is expected by posters on other systems. It would be surprising and alarming for posters to discover late that all their past posts had all been made commercially exploitable by everyone by an obscure TOS clause that nobody reads.

The Discourse people chose CC-BY-NC-SA for their UCL. They didn't state in [their discussion](https://meta.discourse.org/t/who-owns-the-content-you-post-on-a-forum-aka-user-content-licensing/4261/2) why they chose NonCommercial over a Commercial-Allowed version, but I suspect it's for that reason.

IMO: NC would be a guarantee that both ourselves and anyone else aren't going to exploit posters' works without their explicit consent. They'd grant us that by applying a freer license in addition.

> That also includes artwork, screenshots, brushpacks, ect. So if you post your work there without any mention of licence, then it is already eligible to be used on our website and MyPaint itself. If you do not wish to have your artwork under that Creative Commons License but still showcase your artwork, you can do one of three possibilities:

Typo: etc.

> * You can use the Dual Licsence method and just post the low resolution version of your artwork while linking the high quality image under your own licensing.

That's not really what the Dual License thing is about. Posters *have* to grant the site ops the right to republish, and we've selected a specific CC license to allow that. I would suggest:

```
You can post a low-resolution version of your artwork on our site,
while linking to the full quality image under your own licensing.
The full quality image must be hosted on a different site.
```

> * You can also just post a stricter license with the image on the post.

This point needs to be dropped. We *must* be granted the right to republish, so our UCL *must* apply, whatever license we select for the UCL. If a user posts an image in full and tries to state "© Joe Artist, CC BY-NC-ND", then the result is nevertheless dual licensed. Downstream users would be free to select the least restrictive license for reuse/adaptation/redistribution, so slapping an additional stricter license on the content just wouldn't work. And it couldn't make the effective licensing more restrictive.

> * Lastly you can just link the image since the MyPaint Community has no control of third party sites.

This is almost a recap of the first bullet point, given the way Discourse works (doesn't Discourse take an appropriate thumb when it does its fancy links? Fair Use / Fair Dealing applies for thumbnailing off-site links these days, I would guess).

> Do keep in mind that if you go with the later two options, the MyPaint Community team cannot use the artwork on our Website or be distributed with MyPaint.

With an NC license, we wouldn't be able to do that anyway.

-----------------------------

What do you think about making the UCL NonCommercial, just like upstream discourse?

I'd suggest
* not going into the technicalities of DL at all in the Licensing Policy
* retaining the NC module, to avoid surprises
* explaining that if artists want their showcased artwork to be used as promo artwork, they should explicitly provide an (additional) CC-BY-4.0 or CC-BY-SA-4.0 license when posting and allow reposting of their content under that license alone.

**Comment by odysseywestra** [comment]
In my opinion, I would stick to the CC-BY-SA 4.0. The reason I chose the CC-BY-SA 4.0 to allow our content to be distributed with MyPaint and it's website. Plus they are already uploading a lower resolution image because of the file size limits(1mb). Not only is it stated in our TOS, and the FAQ, but also in the topic template in the Artwork Showcase category. So technically users are aware that we're using the CC-BY-SA 4.0.

I flexible with this either direction we go, we just need to make this clear.

**Comment by odysseywestra** [comment]
There is also the option of just have the policy of  [only the original poster may republish contributions](https://meta.discourse.org/t/who-owns-the-content-you-post-on-a-forum-aka-user-content-licensing/4261/3?u=odysseywestra) Since the majority of our users will be artist they may opt for this. That way we can just leave the licensing policy the way it is and just tell if they want their artwork on our website or distributed with MyPaint, they would just need to license their content on our forums as CC-BY or CC-BY-SA. 

A good example of this is the forum for the [KDE Frums](https://forum.kde.org) where the [Krita Community Resides](https://forum.kde.org/viewforum.php?f=136) Their policy clearly states that 
>"All messages belong to their poster and they have the full responsibility for any violation upcoming as a result of their post."
https://forum.kde.org/policy.php#policy4a

I can see this in the long run would be the less headache route than trying to force a user or artist to use a certain license whether it's Creative Commons or the GPL.

**Comment by odysseywestra** [comment]
Deviantart also has a good section about Copyright in its TOS as well.

>DeviantArt is, unless otherwise stated, the owner of all copyright and data rights in the Service and its contents. Individuals who have posted works to DeviantArt are either the copyright owners of the component parts of that work or are posting the work under license from a copyright owner or his or her agent or otherwise as permitted by law. You may not reproduce, distribute, publicly display or perform, or prepare derivative works based on any of the Content including any such works without the express, written consent of DeviantArt or the appropriate owner of copyright in such works. DeviantArt does not claim ownership rights in your works or other materials posted by you to DeviantArt (Your Content). You agree not to distribute any part of the Service other than Your Content in any medium other than as permitted in these Terms of Service or by use of functions on the Service provided by us. You agree not to alter or modify any part of the Service unless expressly permitted to do so by us or by use of functions on the Service provided by us.

**Comment by odysseywestra** [comment]
Okay I've updated the Draft Licence Policy above to reflect the propose changes. Any thoughts?

**Comment by achadwick** [comment]
I've added my own little updates too, covering screenshots submitted via GitHub. That should work similarly to forum postings. The section needed a few other revisions too.

I like keeping the Licensing Policy independent of the forum's choice of UCL, just stating "if you want us to be able to do promo stuff, you gotta license us (and everyone else) to distribute it freely".

**Comment by odysseywestra** [comment]
Alright, Awesome! I'll go ahead and push the changes to our WIki. If we need to do any more changes to our Licensing Policy, just open up a new issue.

**Comment by odysseywestra** [comment]
Alright Changes are pushed on our Wiki.
